core-rust/core-api-server/src/core_api/errors.rs

```diff
@@ -173,6 +175,29 @@ pub(crate) fn assert_unbounded_endpoints_flag_enabled<E: ErrorDetails>(
     Ok(())
 }
 
+impl<E: ErrorDetails> From<PreviewerError> for ResponseError<E> {
```

Comment:
huh, that's a broader topic (since AFAIU it already has more failure modes than "you wanted historical version but this flag isn't enabled").
Also it would be an incompatible schema change, wouldn't it? (actually I am not sure how big of a deal it is, regarding error responses)

Anyways, let me talk to Gateway people first...

Reply:
Let's discuss that offline but in essence introduction of new, strongly-typed (structured) error response should be considered a breaking change.

Reply:
Actually upgrading from a `BasicErrorReponse` to a typed `ErrorResponse` is strictly an extension / backwards compatible. It just adds a `details` field to the error response. `BasicErrorReponse` doesn't have such a field.

Reply:
Still, let's do it in a separate effort: https://radixdlt.atlassian.net/browse/NODE-612